processor/lsmintervalprocessor/processor_test.go

```diff
@@ -352,19 +357,35 @@ func benchmarkAggregation(b *testing.B, ottlStatements []string) {
 			md, err := golden.ReadMetrics(filepath.Join(dir, "input.yaml"))
 			require.NoError(b, err)
 			md.MarkReadOnly()
-			b.ResetTimer()
-
-			err = mgp.Start(context.Background(), componenttest.NewNopHost())
-			require.NoError(b, err)
-			for i := 0; i < b.N; i++ {
+			benchFunc := func() {
+				b.StopTimer()
+				// Copy the metric as the ConsumeMetric call is mutating
 				mdCopy := pmetric.NewMetrics()
 				md.CopyTo(mdCopy)
 				// Overwrites the asdf attribute in metrics such that it becomes high cardinality
-				mdCopy.ResourceMetrics().At(0).Resource().Attributes().PutStr("asdf", fmt.Sprintf("%d", i))
+				mdCopy.ResourceMetrics().At(0).Resource().Attributes().PutStr("asdf", fmt.Sprintf("rand_%d", rand.Int()))
+				b.StartTimer()
+
 				err = mgp.ConsumeMetrics(ctx, mdCopy)
 				require.NoError(b, err)
 			}
 
+			b.ResetTimer()
+			err = mgp.Start(context.Background(), componenttest.NewNopHost())
+			require.NoError(b, err)
+
+			if runParallel {
+				b.RunParallel(func(pb *testing.PB) {
+					for pb.Next() {
+						benchFunc()
```

Comment:
https://pkg.go.dev/testing#B.RunParallel
> The body function will be run in each goroutine. It should set up any goroutine-local state and then iterate until pb.Next returns false. It should not use the [B.StartTimer](https://pkg.go.dev/testing#B.StartTimer), [B.StopTimer](https://pkg.go.dev/testing#B.StopTimer), or [B.ResetTimer](https://pkg.go.dev/testing#B.ResetTimer) functions, because they have global effect. It should also not call [B.Run](https://pkg.go.dev/testing#B.Run).

I'm afraid calling b.StopTimer in a PB won't work

Reply:
might have to bite the bullet and not stop the timer for PB